controller_interface/src/controller_interface.cpp

```diff
@@ -22,12 +22,14 @@ ControllerInterface::ControllerInterface() : ControllerInterfaceBase() {}
 
 bool ControllerInterface::is_chainable() const { return false; }
 
-std::vector<hardware_interface::StateInterface> ControllerInterface::export_state_interfaces()
+std::vector<std::shared_ptr<hardware_interface::StateInterface>>
+ControllerInterface::export_state_interfaces()
 {
   return {};
 }
 
-std::vector<hardware_interface::CommandInterface> ControllerInterface::export_reference_interfaces()
+std::vector<std::shared_ptr<hardware_interface::CommandInterface>>
```

Comment:
no because then they would not override the base functions (which have to have this return type because controllers need access to the exported handles to set/get values from them).

Reply:
I think @bmagyar means here that we define a new type, something like:

```
using CommandInterfacePtr = std::shared_ptr<CommandInterface>
```

This should go to the header where `CommandInterface` is defined. 

On the other ahdn I am not fan of it, as it makes things shorter, but make them also less  clear and direct.